src/lib/y2users/password.rb

```diff
@@ -93,5 +103,46 @@ def ==(other)
       # do not compare configuration to allow comparison between different configs
       ATTRS.all? { |a| public_send(a) == other.public_send(a) }
     end
+
+    # Represents password value. It specific type is defined as subclass and can be queried
+    class Value
```

Comment:
I don't like to have nested classes when we intent to use them from the outside world.

In other words, having a class `Password` and then a class `Password::Value` is fine if the inner one is only known to `Password`. But since we have a `Password#value` method that returns an object of the `Value` class, I would rather have `Password` and `PasswordValue`.

At least, that's the criterion we have followed in other parts like storage-ng.

Reply:
I have no strong opinion about this. In general, I like nested class names when it makes sense, even to use them publicly. But if we are already following a criterion, then I would respect it as much as possible.

Reply:
I have no strong opinion for this concrete case either, but I tend to avoid nesting. So, I prefer `PasswordValue` over `Password::Value`.

Reply:
What I prefer about nested names is that `Password::Value` states that `Value` is something that only makes sense in the context of `Password`, I mean, it is something to be used for a password and nothing else. Otherwise, `PasswordValue` is only a name, and nothing indicates that it is strongly attached to a `Password`. But again, a matter of taste.

Reply:
Just to clarify, I don't have such a strong opinion either and I also see advantages in the nested approach. But I remember that we agreed in the rule of "nested only if not visible from the outside" back then in the context of storage-ng and I think it was intended as a general rule for all the new naming. But I can be wrong and maybe the rule does not apply here.

@imobachgs Do you remember better?

Reply:
From https://gist.github.com/imobachgs/d90940315bd0471ee00b2a68ba22fbe5#filesnamespaces-organization I can read that nested classes are:

* Only for private APIs.
* Avoid cases like the storage Proposal code when possible.

I do not have a strong opinion. I tend to avoid nested classes but just because I do not want to end up with a 2000 LoC file containing a lot of stuff. However, I am fine with keeping ancillary (*and simple*) classes nested.

About the `Password::Value` vs `PasswordValue` discussion, it is secondary to me. I am sure that we can find many classes whose instances are only relevant when combined with another specific class, and they are not nested.

Finally, if the `Password::Value` class is simple enough and we do not plan to extend it in the future, just keep it nested. Otherwise, I would go for a separate one.

Reply:
> Otherwise, `PasswordValue` is only a name, and nothing indicates that it is strongly attached to a `Password`

That's true